src/geometry/algebra/lie_group.lean

```diff
@@ -0,0 +1,258 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Nicolò Cavalleri.
+-/
+
+import geometry.manifold.times_cont_mdiff
+
+noncomputable theory
+
+/-!
+# Lie groups
+
+We define Lie groups.
+
+## Main definitions and statements
+
+* `lie_add_group I G` : a Lie additive group where `G` is a manifold on the model with corners `I`.
+* `lie_group I G`     : a Lie multiplicative group where `G` is a manifold on the model with
+                        corners `I`.
+
+## Implementation notes
+A priori, a Lie group here is a manifold with corner.
```

Comment:
This is the most important part of the file, so it should be expanded a lot, adding at least the following:
* In the first lines, you should say what a Lie group is in mathematical terms.
* In the implementation notes, you should explain why one can not take just the identity model with corner on a vector space, due to issues with products.
* In the main definitions and statements, you should mention morphisms and of_core, and say that you prove that reals are a Lie group.